console/network/src/lib.rs

```diff
@@ -410,3 +449,132 @@ pub trait Network:
         leaf: &Vec<Field<Self>>,
     ) -> bool;
 }
+
+/// Returns the consensus configuration value for the specified height.
+///
+/// Arguments:
+/// - `$network`: The network to use the constant of.
+/// - `$constant`: The constant to search a value of.
+/// - `$seek_height`: The block height to search the value for.
+#[macro_export]
+macro_rules! consensus_config_value {
+    ($network:ident, $constant:ident, $seek_height:expr) => {
+        // Search the consensus version enacted at the specified height.
+        $network::CONSENSUS_VERSION($seek_height).map_or(None, |seek_version| {
+            // Search the consensus value for the specified version.
+            match $network::$constant.binary_search_by(|(version, _)| version.cmp(&seek_version)) {
+                // If a value was found for this consensus version, return it.
+                Ok(index) => Some($network::$constant[index].1),
+                // If the specified version was not found exactly, determine whether to return an appropriate value anyway.
+                Err(index) => {
+                    // This constant is not yet in effect at this consensus version.
+                    if index == 0 {
+                        None
+                    // Return the appropriate value belonging to the consensus version *lower* than the sought version.
+                    } else {
+                        Some($network::$constant[index - 1].1)
+                    }
+                }
+            }
+        })
+    };
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    /// Ensure that the consensus constants are defined and correct at genesis.
+    /// It is possible this invariant no longer holds in the future, e.g. due to pruning or novel types of constants.
+    fn consensus_constants_at_genesis<N: Network>() {
+        let height = N::CONSENSUS_VERSION_HEIGHTS.first().unwrap().1;
+        assert_eq!(height, 0);
+        let consensus_version = N::CONSENSUS_VERSION_HEIGHTS.first().unwrap().0;
+        assert_eq!(consensus_version, ConsensusVersion::V1);
+        assert_eq!(consensus_version as usize, 1);
+    }
+
+    /// Ensure that the consensus *versions* are unique, incrementing and start with 0.
+    fn consensus_versions<N: Network>() {
+        let mut previous_version = N::CONSENSUS_VERSION_HEIGHTS.first().unwrap().0;
+        // Ensure that the consensus versions start with 0.
```

Comment:
```suggestion
        // Ensure that the consensus versions start with 1.
```
unless what is meant here is the height

Reply:
Thx 4beaa7a82